### PR TITLE
Copy and adjust vivarium modeling strategy from sepsis to OL page

### PR DESCRIPTION
**Pull request by pletale**
I don't think the graphviz diagram is rendering properly - I tried installing python_graphviz to my conda environment but locally the diagram isn't rendering. This is all copied over from the sepsis page with a few tweaks.

### CODE REVIEW

**Comment by zmbc** [comment]
> I tried installing python_graphviz to my conda environment

You need `conda install graphviz`, _not_ only python_graphviz.

**Comment by pletale** [comment]
> > I tried installing python_graphviz to my conda environment
> 
> You need `conda install graphviz`, _not_ only python_graphviz.

Ah okay. I tried installing with conda at first but got an SSL error (I think I might have an idea why?), so I tried installing with pip which seemed to make the build error go away. Glad to know the diagram is rendering properly in the remote version!

**Comment by zmbc** [comment]
I don't think installing with pip should fix things, as that just installs the Python package, not the underlying Graphviz tool. But glad it is working for you now!

File: docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst [hunk 1]
```diff
@@ -113,11 +113,105 @@ Vivarium Modeling Strategy
...
+while dashed lines indicate pieces of the underlying pregnancy model.
+
+.. graphviz::
+
+    digraph OL_decisions {
+        rankdir = LR;
+        ftp [label="full term\npregnancy, post\nintrapartum", style=dashed]
+        ftb [label="full term\nbirth", style=dashed]
+        alive [label="parent alive"]
+        dead [label="parent dead"]
```
Comment (pletale) [inline docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst:165]: In PR [#1554](https://github.com/ihmeuw/vivarium_research/pull/1554#issue-2680986294) I changed these labels to 'parent did not die of cause' and 'parent died of cause' to be more accurate to our simulation.

File: docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst [hunk 2]
```diff
@@ -113,11 +113,105 @@ Vivarium Modeling Strategy
 Scope
 +++++
 
+The goal of the obstructed labor model is to capture YLLs and YLDs due to
+obstructed labor and uterine rupture among women of
+reproductive age. We only model obstructed labor and uterine rupture among 
+simulants who give (live or still) birth after a full term pregnancy. This 
+page documents how to model the baseline burden of obstructed labor and 
+uterine rupture. Other simulation components such as c-sections will affect 
+the rates of the pages for the corresponding :ref:`intervention <intervention_models>`
```
Comment (NathanielBlairStahn) [inline docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst:122]:
```suggestion
the rates of obstructed labor; the pages for the corresponding :ref:`intervention <intervention_models>`
```

File: docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst [hunk 3]
```diff
@@ -113,11 +113,105 @@ Vivarium Modeling Strategy
...
+    * - State
+      - Definition
+    * - full term pregnancy, post intrapartum
+      - Parent simulant has a full term pregnancy as determined by the
+        :ref:`pregnancy model
+        <other_models_pregnancy_closed_cohort_mncnh>`, **and** has
+        already been through the antenatal and intrapartum models
+    * - OL
+      - Parent simulant experiences obstructed labor or uterine rupture 
+        (fistula).
```
Comment (NathanielBlairStahn) [inline docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst:188]: Why does this say "fistula" in parentheses?

File: docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst (hunk 1 above)
Comment (NathanielBlairStahn) [reply docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst:165]: FYI @pletale: I see that you branched https://github.com/ihmeuw/vivarium_research/pull/1554 off of this branch, but generally it's easier to avoid merge conflicts if you always branch new PRs off the `main` branch and edit completely disjoint sections of the page in each PR. (Granted, that's not always possible, but in this case it is.)

For example, if you make an edit in this PR based on feedback, then you'll have to merge the new edits from this PR into PR 1554 before that PR can be merged, which can lead to conflicts that need to be resolved, potentially introducing new errors (I've seen this happen before, requiring a follow-up PR to correct the merge errors). I added section headers to each page so that each PR can correspond to content between two section headers, which will ensure that the content in different PRs is disjoint, as long as the headers themselves are not edited (this was James Collins's suggested strategy for editing the docs).

In other words, it would probably be easier to make all the changes to the "Vivarium Modeling Strategy" and "Cause Model Decision Graph" sections in this PR, and in the other PR, _only_ make changes to the "Data Tables" section. So I would recommend _closing_ https://github.com/ihmeuw/vivarium_research/pull/1554, and moving the edits to the to the graph and to the "Scope" section to this PR instead, and copy all the edits to the "Data Tables" section into a _new_ PR that is branched off of `main`.

@zmbc @aflaxman @SylLutze Does this seem like generally good practice to you (i.e. keeping content in different PRs disjoint, and branching off `main` when possible)? Do any of you have any other suggestions for making things easy and avoiding merge conflicts?

Comment (zmbc) [reply docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst:165]: I don't think the branching strategy has too much impact on the likelihood of merge conflicts -- as long as you have the section headers as you mention, you really shouldn't get conflicts. But generally I do think it is good to make separate PRs separate (not on top of each other) so they could be merged in any order.

File: docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst [hunk 4]
```diff
@@ -113,11 +113,105 @@ Vivarium Modeling Strategy
...
+    * - parent alive
+      - Parent simulant is still alive
```
Comment (NathanielBlairStahn) [inline docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst:190]: This state name and definition will need to get updated with whatever new names we settle on.

File: docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst [hunk 5]
```diff
@@ -113,11 +113,105 @@ Vivarium Modeling Strategy
...
+    * - parent dead
```
Comment (NathanielBlairStahn) [inline docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst:191]: This state name should get updated to match the updated graph.

File: docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst (hunk 1 above)
Comment (NathanielBlairStahn) [reply docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst:165]: > as long as you have the section headers as you mention, you really shouldn't get conflicts

In this case, https://github.com/ihmeuw/vivarium_research/pull/1554 is making edits to the sections in this PR, so I'm more confident that merge conflicts will be avoided if we simply create a new PR off of `main` after moving those edits to this PR, rather than try to revert those edits in the other PR.

> so they could be merged in any order

Ah, yes, that's another good reason to branch off `main` when possible, so that it doesn't get confusing when you try to merge everything. I've seen that happen, too, where merging in the wrong order required a follow-up PR to correct things: https://github.com/ihmeuw/vivarium_research/pull/1530

File: docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst [hunk 6]
```diff
@@ -113,11 +113,104 @@ Vivarium Modeling Strategy
...
+    * - State
+      - Definition
+    * - full term pregnancy, post intrapartum
+      - Parent simulant has a full term pregnancy as determined by the
+        :ref:`pregnancy model
+        <other_models_pregnancy_closed_cohort_mncnh>`, **and** has
+        already been through the antenatal and intrapartum models
+    * - OL
+      - Parent simulant experiences obstructed labor or uterine rupture
+    * - parent not dead of OL
```
Comment (pletale) [inline docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst:188]: @NathanielBlairStahn I updated these state names and definitions but not sure if the names are good!

File: docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst [hunk 7]
```diff
@@ -113,11 +113,104 @@ Vivarium Modeling Strategy
 Scope
 +++++
 
+The goal of the obstructed labor (OL) model is to capture YLLs and YLDs due to
+obstructed labor and uterine rupture among women of
+reproductive age. We only model obstructed labor and uterine rupture among 
+simulants who give (live or still) birth after a full term pregnancy. This 
+page documents how to model the baseline burden of obstructed labor and 
+uterine rupture. Other simulation components such as c-sections will affect 
+the rates of obstructed labor; the pages for the corresponding :ref:`intervention <intervention_models>`
```
Comment (NathanielBlairStahn) [inline docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst:122]:
```suggestion
the rates of obstructed labor; such effects will be described on the pages for the corresponding :ref:`intervention <intervention_models>`
```

File: docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst (hunk 6 above)
Comment (NathanielBlairStahn) [reply docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst:188]:
```suggestion
    * - parent did not die of OL 
```
Looks good! But I think we should match the wording in the diagram exactly.

File: docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst [hunk 8]
```diff
@@ -113,11 +113,104 @@ Vivarium Modeling Strategy
...
+lines are the pieces added by the obstructed labor and uterine rupture model, 
+while dashed lines indicate pieces of the underlying pregnancy model.
+
+.. graphviz::
+
+    digraph OL_decisions {
+        rankdir = LR;
+        ftp [label="full term\npregnancy, post\nintrapartum", style=dashed]
+        ftb [label="full term\nbirth", style=dashed]
+        alive [label="parent did not die of OL"]
```
Comment (NathanielBlairStahn) [inline docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst:164]:
```suggestion
        alive [label="parent did not\ndie of OL"]
```
Let's try splitting it into two lines to make the diagram less wide

File: docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst [hunk 9]
```diff
@@ -113,11 +113,104 @@ Vivarium Modeling Strategy
...
+        dead [label="parent died of OL"]
```
Comment (NathanielBlairStahn) [inline docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst:165]:
```suggestion
        dead [label="parent died\nof OL"]
```
ditto

File: docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst [hunk 10]
```diff
@@ -113,11 +113,104 @@ Vivarium Modeling Strategy
...
+    * - Symbol
+      - Name
+      - Definition
+    * - ir
+      - incidence risk
+      - The probability that a pregnant simulant experiences obstructed 
+        labor or uterine rupture
+    * - cfr
+      - case fatality rate
+      - The probability that a simulant with  experiences dies of that
```
Comment (NathanielBlairStahn) [inline docs/source/models/causes/maternal_disorders/gbd_2021_mncnh/obstructed_labor.rst:211]:
```suggestion
      - The probability that a simulant who experiences obstructed labor or uterine rupture dies of that 
```

**Review by NathanielBlairStahn — APPROVED** [review]
Looks good except for a few minor grammatical and formatting fixes.
